### PR TITLE
MODLD-333: Assign correct ID for work resource

### PR DESCRIPTION
**Pull request by pkjacob**

### CODE REVIEW

File: src/test/java/org/folio/marc4ld/mapper/Marc2BibframeMapperIT.java [hunk 1]
```diff
@@ -1117,4 +1120,8 @@ private void validateSubjectEdge(ResourceEdge subjectEdge, List<ResourceTypeDict
     assertThat(edgeIterator.hasNext()).isFalse();
   }
 
+  private void validateId(Resource resource) {
+    var expectedId = hashService.hash(resource);
+    assertThat(resource.getId()).isEqualTo(expectedId);
```
Comment (pkjacob) [inline src/test/java/org/folio/marc4ld/mapper/Marc2BibframeMapperIT.java:1125]: Here we are asserting that the application invoked ` hashService.hash(resource)` after setting all the properties of the resource.

File: src/test/java/org/folio/marc4ld/service/marc2ld/postprocessor/InstanceResourcePostProcessorTest.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+  private Resource createResource(Long id, JsonNode doc, Resource... edges) {
+    var resource = new Resource().setId(id).setDoc(doc);
+    Stream.of(edges)
+      .forEach(e -> createConnection(resource, e, EDITOR));
+    return resource;
+  }
+
+  private void createConnection(Resource from, Resource to, PredicateDictionary predicate) {
+    from.getOutgoingEdges().add(new ResourceEdge(from, to, predicate));
```
Comment (pkjacob) [inline src/test/java/org/folio/marc4ld/service/marc2ld/postprocessor/InstanceResourcePostProcessorTest.java:110]: Minor - Looks like we need to add an `addOutgoingEdge(..)` method in the Resource class.

Comment (PBobylev) [reply src/test/java/org/folio/marc4ld/service/marc2ld/postprocessor/InstanceResourcePostProcessorTest.java:110]: sure, why not!

**Review by PBobylev — APPROVED** [review]

**Comment by pkjacob** [comment]
Closing this PR. This PR was merged to fix this problem - https://github.com/FOLIO-EIS/lib-linked-data-marc4ld/pull/87

**Pull request closed without merging**